pkg/rbac/sharedData.go

```diff
@@ -51,36 +55,108 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
-func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+func (shared *SharedData) getPropertyTypes(cache *Cache, ctx context.Context) (map[string]string, error) {
+
+	// original query: select distinct key, jsonb_typeof(value) as datatype FROM search.resources,jsonb_each(data);
+	var selectDs *goqu.SelectDataset
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct()
+
+	query, params, sqlerr := selectDs.ToSQL()
+	if sqlerr != nil {
+		klog.Errorf("Error building Search query: %s", sqlerr.Error())
+	}
+
+	klog.V(5).Infof("Query for property datatypes: [%s] ", query)
+	rows, err := cache.pool.Query(ctx, query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	resourceTypeMap := make(map[string]string)
+	for rows.Next() {
+		var key, value string
+		err = rows.Scan(&key, &value)
+
+		if err != nil {
+			klog.Errorf("Error %s retrieving rows for query:%s", err.Error(), query)
+			continue
+		}
+		resourceTypeMap[key] = value
+
+	}
+	//cache query:
+	shared.propTypes = resourceTypeMap
+	shared.propTypeErr = err
+	shared.propTypeTime = time.Now()
+
+	return resourceTypeMap, err
+}
+
+func (cache *Cache) GetSharedData(ctx context.Context) (map[string]string, error) {
+	sharedData, err := cache.PopulateSharedCache(ctx)
+	if err != nil {
+		klog.Error("Error populating shared data cache: ", err)
+		return nil, err
+	}
+	//store only the PropTypeCache to use in outside of rbac module
+	SharedDataAccess := sharedData.GetPropertyTypeCache()
+
+	propTypesMap := SharedDataAccess
+	return propTypesMap, nil
+}
+
+func (shared *SharedData) GetPropertyTypeCache() map[string]string {
+	return shared.propTypes
+}
```

Comment:
Do we need this function? I only see 1 reference, which could be refactored without the additional function.
<img width="734" alt="image" src="https://user-images.githubusercontent.com/4671325/193639316-16efd6a9-8112-4090-92ce-6729ac4fc69c.png">
```suggestion
	//store only the PropTypeCache to use in outside of rbac module
	propTypesMap := shared.propTypes
	return propTypesMap, nil
}
```

Reply:
Done.